keras/layers/core.py

```diff
@@ -1694,3 +1795,92 @@ def add_shared_layer(layer, inputs):
         sh = SiameseHead(i)
         inputs[i].add(s)
         inputs[i].add(sh)
+
+class Highway(Layer):
+    '''Densely connected highway network, 
+    a natural extension of LSTMs to feedforward networks
+
+    This layer has identical output shape to input shape, and
+    thus doesn't need an output_dim specified
+
+    cite: http://arxiv.org/pdf/1505.00387v2.pdf
+    '''
+    input_ndim = 2
+
+    def __init__(self, init='glorot_uniform', transform_bias=-2, activation='linear', weights=None,
+                 W_regularizer=None, b_regularizer=None, activity_regularizer=None,
+                 W_constraint=None, b_constraint=None, input_dim=None, **kwargs):
+        self.init = initializations.get(init)
+        self.transform_bias = transform_bias
+        self.activation = activations.get(activation)
+
+        self.W_regularizer = regularizers.get(W_regularizer)
+        self.b_regularizer = regularizers.get(b_regularizer)
+        self.activity_regularizer = regularizers.get(activity_regularizer)
+
+        self.W_constraint = constraints.get(W_constraint)
+        self.b_constraint = constraints.get(b_constraint)
+        self.constraints = [self.W_constraint, self.b_constraint]
+
+        self.initial_weights = weights
+
+        self.input_dim = input_dim
+        if self.input_dim:
+            kwargs['input_shape'] = (self.input_dim,)
+        self.input = K.placeholder(ndim=2)
+        super(Highway, self).__init__(**kwargs)
+
+    def build(self):
+        input_dim = self.input_shape[1]
+
+        self.W = self.init((input_dim, input_dim))
+        self.W_carry = self.init((input_dim, input_dim))
+        
+        self.b = K.zeros((input_dim,))
+        # -- initialize with a vector of values `transform_bias`
+        self.b_carry = K.variable(np.ones((input_dim,)) * self.transform_bias)
+
+        self.params = [self.W, self.b, self.W_carry, self.b_carry]
+
+        self.regularizers = []
+        if self.W_regularizer:
+            self.W_regularizer.set_param(self.W)
+            self.regularizers.append(self.W_regularizer)
+
+        if self.b_regularizer:
+            self.b_regularizer.set_param(self.b)
+            self.regularizers.append(self.b_regularizer)
+
+        if self.activity_regularizer:
+            self.activity_regularizer.set_layer(self)
+            self.regularizers.append(self.activity_regularizer)
+
+        if self.initial_weights is not None:
+            self.set_weights(self.initial_weights)
+            del self.initial_weights
+
+    @property
+    def output_shape(self):
+        return (self.input_shape[0], self.input_shape[1])
+
+    def get_output(self, train=False):
+        X = self.get_input(train)
+        transform_weight = activations.sigmoid(K.dot(X, self.W_carry) + self.b_carry)
```

Comment:
In general I suppose so, but there is the problem of `(a, (1-a))` needing to define a simplex. In the paper, they use sigmoid. Think it would be useful to allow people to use `hard_sigmoid` or `ultra_fast_sigmoid`?